controllers/neutronapi_controller.go

```diff
@@ -491,7 +491,43 @@ func (r *NeutronAPIReconciler) reconcileInit(
 			err.Error()))
 		return ctrl.Result{}, err
 	}
-	secretVars[tls.TLSHashName] = env.SetValue(certsHash)
+
+	if apiCertsHash != "" {
+		secretVars["apiCertsHash"] = env.SetValue(apiCertsHash)
+	}
+
+	// Validate OVN service cert secret
+	if instance.Spec.TLS.Ovn.Enabled() {
+		ovnCertsHash, err := instance.Spec.TLS.Ovn.ValidateCertSecret(ctx, helper, instance.Namespace)
+		if err != nil {
+			if k8s_errors.IsNotFound(err) {
+				instance.Status.Conditions.Set(condition.FalseCondition(
+					condition.TLSInputReadyCondition,
+					condition.RequestedReason,
+					condition.SeverityInfo,
+					fmt.Sprintf(condition.TLSInputReadyWaitingMessage, err.Error())))
+				return ctrl.Result{}, nil
+			}
+			instance.Status.Conditions.Set(condition.FalseCondition(
+				condition.TLSInputReadyCondition,
+				condition.ErrorReason,
+				condition.SeverityWarning,
+				condition.TLSInputErrorMessage,
+				err.Error()))
+			return ctrl.Result{}, err
+		}
+
+		if ovnCertsHash != "" {
+			secretVars["ovnCertsHash"] = env.SetValue(ovnCertsHash)
+		}
+	}
+
+	// Combine all cert hashes into a single hash
+	combinedCertsHash, err := util.HashOfInputHashes(secretVars)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	secretVars[tls.TLSHashName] = env.SetValue(combinedCertsHash)
```

Comment:
I don't think we need this. bellow in L539 there is a call for `r.createHashOfInputHashes()` which calcs the hash of hashes

Reply:
Right, I thought about keeping the original logic with `tls.TLSHashName` but it's redundant to calculate hashes two times.